src/core/lib/layer.js

```diff
@@ -610,29 +616,41 @@ export default class Layer {
   }
 
   _updateBaseUniforms() {
-    this.setUniforms({
+    const uniforms = {
       // apply gamma to opacity to make it visually "linear"
       opacity: Math.pow(this.props.opacity, 1 / 2.2),
       ONE: 1.0
-    });
+    };
+    for (const model of this.getModels()) {
+      model.setUniforms(uniforms);
+    }
+
+    // TODO - set needsRedraw on the model(s)?
+    this.state.needsRedraw = true;
   }
 
-  _updateModuleSettings() {
-    if (this.state.model) {
-      this.state.model.updateModuleSettings({
-        pickingHighlightColor: this.props.highlightColor
-      });
+  _updateBaseModuleSettings() {
+    const settings = {
+      pickingHighlightColor: this.props.highlightColor
+    };
+    this.updateModuleSettings(settings);
+  }
+
+  updateModuleSettings(settings) {
```

Comment:
Is `updateModuleSettings` an external API or a private method? Also, what are the reasons to have two similar methods here (`_updateBaseModuleSettings` and `updateModuleSettings`)?

Reply:
`_updateBaseModuleSettings` sets default module settings based on props, similar to what `_updateBaseUniforms` do. Per @ibgreen 's comment above, we may not want to keep `updateModuleSettings` and `setUniforms` as public methods, if we prefer external code directly accessing `getModels`.